rust/src/proofs/api.rs

```diff
@@ -421,16 +421,17 @@ pub unsafe extern "C" fn fil_unseal_range(
 
         info!("unseal_range: start");
 
+        use filepath::FilePath;
         use std::os::unix::io::{FromRawFd, IntoRawFd};
 
-        let mut sealed_sector = std::fs::File::from_raw_fd(sealed_sector_fd_raw);
-        let mut unseal_output = std::fs::File::from_raw_fd(unseal_output_fd_raw);
+        let sealed_sector = std::fs::File::from_raw_fd(sealed_sector_fd_raw);
+        let unseal_output = std::fs::File::from_raw_fd(unseal_output_fd_raw);
 
-        let result = filecoin_proofs_api::seal::unseal_range(
+        let result = filecoin_proofs_api::seal::get_unsealed_range(
             registered_proof.into(),
             c_str_to_pbuf(cache_dir_path),
-            &mut sealed_sector,
-            &mut unseal_output,
+            sealed_sector.path().unwrap(),
```

Comment:
these might be pipes, I don' tthink there is a gurantee that there will always be a filepath available.

Reply:
Yes, I'm aware.  Unfortunately, a caller re-factor or audit is also required to make use of this.

Reply:
`sealed_sector` is currently passed as a normal file

Reply:
Great, thanks!  What about the unseal output?  Right now I am also assuming it's a normal file, but there are options to work around that if we have to.

Reply:
Unseal output is always an anonymous pipe - on the Go side it's just piped into a fr32 padded file with a trailer containing an rle which marks which parts of the file are allocated

See https://github.com/filecoin-project/lotus/blob/b54b1303f9f6bcc3b57ae25cfc1174b59f44b943/extern/sector-storage/ffiwrapper/sealer_cgo.go#L322-L362